mods/farming/api.lua

```diff
@@ -189,6 +189,15 @@ farming.place_seed = function(itemstack, placer, pointed_thing, plantname)
 	return itemstack
 end
 
+-- check if on wet soil
+farming.can_grow = function(pos)
+	local below = minetest.get_node({x = pos.x, y = pos.y - 1, z = pos.z})
```

Comment:
"new" way to write this would be `pos:offset(0, -1, 0)` or `vector.new(pos.x, pos.y - 1, pos.z)` if you want to code defensively